File: app/src/main/java/com/tcssol/expensetracker/Utils/Converters.java
package com.tcssol.expensetracker.Utils;
import androidx.room.TypeConverter;
import java.time.LocalDate;
import java.time.format.DateTimeFormatter;

public class Converters {

    private static final DateTimeFormatter formatter = DateTimeFormatter.ISO_LOCAL_DATE;

    @TypeConverter
    public static LocalDate fromString(String value) {
        return value == null ? null : LocalDate.parse(value, formatter);
    }

    @TypeConverter
    public static String toString(LocalDate date) {
        return date == null ? null : date.format(formatter);
    }
}

File: app/src/main/java/com/tcssol/expensetracker/Utils/Converters.kt
package com.tcssol.expensetracker.Utils

import androidx.room.TypeConverter
import java.time.LocalDate
import java.time.format.DateTimeFormatter

class Converters {
    companion object {
        private val formatter: DateTimeFormatter = DateTimeFormatter.ISO_LOCAL_DATE

        @JvmStatic
        @TypeConverter
        fun fromString(value: String?): LocalDate? {
            return value?.let { LocalDate.parse(it, formatter) }
        }

        @JvmStatic
        @TypeConverter
        fun toString(date: LocalDate?): String? {
            return date?.format(formatter)
        }
    }
}
